spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1662,8 +1652,7 @@ def _do_mapping(self, run_time, total_run_time):
 
         # only add the partitioner if there isn't already a machine graph
         algorithms.append("MallocBasedChipIDAllocator")
-        if (self._application_graph.n_vertices and
-                not self._machine_graph.n_vertices):
+        if self._application_graph.n_vertices:
```

Comment:
The comment above this says to only add the partitioner if there isn't already a machine graph.  I am guessing this is from the "re-run" mode of operation i.e. a run has already happened, the application graph hasn't changed and so the machine graph doesn't need to be re-created.  Removing this check might mean that a re-run causes a re-partition.

Reply:
Yes. Changed the check to just if not self._machine_graph.n_vertices:
We know that from should_run if the machine_graph has not vertexes the application one must.

Reply:
Should_run checks the original graphs though, not the clones.  This specific check is testing if partitioning should happen; it shouldn't if the graphs haven't changed and you are re-running.

Reply:
I might be getting the path wrong here though; this is in _do_mapping, so if this has already decided to do all mapping, maybe redoing partitioning here is correct?

Reply:
The check is now:
if not self._machine_graph.n_vertices:

This confirms if partitioning is needed or not,
The second check self._application_graph.n_vertices was to avoid partitioning and empty graph. Should run has already taken care of the empty graphs.

Reply:
OK, sounds good